### PR TITLE
UI: Add confirmation dialog if there are no sources

### PR DESCRIPTION
**Pull request by cg2121**
This adds a confirmation dialog for streaming or recording if
there are no sources.

### CODE REVIEW

File: UI/data/locale/en-US.ini [hunk 1]
```diff
@@ -845,3 +845,8 @@ OutputWarnings.MP4Recording="Warning: Recordings saved to MP4 will be unrecovera
 # deleting final scene
 FinalScene.Title="Delete Scene"
 FinalScene.Text="There needs to be at least one scene."
+
+# no sources
+NoSources.Title="No Sources"
+NoSources.Text.Stream="There are no sources! Are you sure you want to stream a blank scene?"
```
Comment (WizardCM) [inline UI/data/locale/en-US.ini]: I'd change it to something friendlier, and more helpful.

"It looks like you haven't added any sources to your scene, so you will only be recording a blank screen. Are you sure you want to do this?

You can add sources by clicking the + under the Sources box in the main window at any time."

**Comment by cg2121** [comment]
I have now changed the wording to your suggestion.

**Comment by jp9000** [comment]
There's currently multiple lines of thought here, so I'm delaying this for now.

**Comment by jp9000** [comment]
So the idea I have is instead of displaying a dialog, we replace the contents of the list box with a big message that asks users to add sources if they don't already have any.  But..  hmm, would this necessarily conflict with the dialog?  Maybe I might still merge it.

**Comment by jp9000** [comment]
Okay, only one suggestion:  add an option to allow the user to turn this off, but have this on by default.  Then it should be good

**Comment by jp9000** [comment]
Actually, I'm kind of flip-flopping here on this one.

For scenes that are empty, I feel like this really should not be necessary.  I feel like we should have a graphical display, I think in the source list widget, or replacing the source list widget, that explains this to the user without necessarily warning them.

However, we *should* have this dialog if the user has absolutely no non-scene sources.  For example, They have one or more scenes, but all of them are empty.  If that's the case, then it should absolutely display this dialog, every time, and furthermore, there's no need to allow the user to disable it like I had almost suggested, because it doesn't make sense for the user to record/stream if they have nothing added anywhere.

**Comment by jp9000** [comment]
So to sum up my view:

1.) If the currently displayed scene is an empty scene, the source list should show text and/or image or something, allowing the user to click on it to add sources directly.  It's be something like `This scene is empty.  Click the '+ button or right-click to add sources`.  Some of the others have suggested this before and it's a good idea.  After they add a source, the source list will revert to its normal functionality and show as a list.  This way, the user knows they have to add sources to display anything.  It'd be nicer from a design perspective.

2.) If all of the user's scenes are empty, *then* we should absolutely display this dialog each time, 100%.

I'm not sure how long it would take to make those changes however.  If you'd like, just implement this as item number 2 there, then I can help fill in item number 1.

**Comment by jp9000** [comment]
Actually, you already do it that way -- perfect, I'll just merge the PR as-is.  Sorry for all the back-and forth rambling there -- turns out it was good as-is and operated the way I'd hoped.

**Comment by cg2121** [comment]
I've now added the 'Do not show again' checkbox to the dialog, and fixed a couple bugs that I have found.

**Pull request closed without merging**